src/Processors/Executors/PipelineExecutor.cpp

```diff
@@ -432,6 +432,11 @@ bool PipelineExecutor::executeStep(std::atomic_bool * yield_flag)
         if (node->exception)
             std::rethrow_exception(node->exception);
 
+    /// Exception which happened in executing thread, but not at processor.
+    for (auto & executor_context : executor_contexts)
+        if (executor_context->exception)
+            std::rethrow_exception(executor_context->exception);
+
```

Comment:
It seems `executeStepImpl()` never set `executor_context->exception`, so this cycle doesn't look useful.

Reply:
Right. Wring place.